### ISSUE TITLE
Installation experience could be improved

### ISSUE THREAD

**Issue by lmeuser** [cafca/souma#177]
I just tried to install Souma on my machine. I could not find any installation instructions -- there are two readmes, but one of them only says I should sign up for the beta, and the other contains incomplete and wrong instructions -- it says nothing about dependency installation and tells me to run a non-existing file (set_hosts.py).

So, I just set up a virtualenv, cloned the repo into it, and tried pip install -e souma — this also failed, because some of the dependencies wouldn’t compile until I added “-Qunused-arguments” to my CFLAGS (this seems to be an issue with the other libraries on the current OS X version and not a souma issue, but an installation guide could mention this) and installed a Fortran compiler (???).
Then it installed the dependencies correctly, but the installation souma itself failed with “error: <virtualenv path>/souma/set_hosts.py: No such file or directory”. It still seems to have installed souma partly, but trying to run the installed run.py resulted in server errors because it could not find template files. Running it from the repository instead of the installed version seems to work.

So, to sum up, I think that
- the setup.py and related files should be fixed to allow installation using python’s normal packaging tools and procedures
- there should be a step-by-step installation guide explaining what exactly one needs to do to get souma up and running (and preferably why the over 100MB of dependencies that take almost half an hour to build [on my machine] and require stuff like a Fortran compiler are really necessary)

Even if the plan is that people simply use binary packages and never deal with these issues, this would still make the project more accessible for developers.

**Comment by cafca** [comment]
Ok, I see we can do a lot better with documenting the build process. I already fixed some small things in the [build instructions](https://github.com/ciex/souma/wiki/Development-Environment) yesterday but there is more to do. I didn't know about the issue with set_hosts and especially the templates. Can you paste the error log from when it couldn't find them?

Most of the other problems come from using numpy/scipy - all those dependencies, Fortran, etc. Maybe we can implement the parts we use from those libraries ourselves, I will talk to @kahartma about that.

**Comment by lmeuser** [comment]
Ah, for some reason it did not occur to me to look for installation instructions in the wiki, my bad. Still, I think a copy of those or a pointer to that page in the repo itself (either one of the README file, or a separate INSTALL file) would be helpful.

The set_hosts.py issue is simple -- the installation wiki page, setup.py and README all reference a file called set_hosts.py in the root directory of the repository, which does not exist (there is one in nucleus/, but it doesn't appear to do anything if called from the command line).

Tracebacks for files that are not found: https://gist.github.com/lmeuser/f8f97866da599938453a
The first one happens during startup (followed by #182), the second one is the missing template and happens when I open the /login URL of the web app. /Users/lukas/code/souma is my souma venv. I run the installed-by-pip version of run.py from that directory. /Users/lukas/code/souma/souma is my clone of the repo. The first bug seems to be because it looks for a file relative to the working directory instead of the installation directory, I suspect the template lookup does the same.
